core/src/main/java/com/linecorp/armeria/common/outlier/OutlierDetection.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.outlier;
+
+import static com.linecorp.armeria.common.outlier.OutlierDetectionBuilder.DEFAULT_DETECTION;
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * An outlier detection provides a mechanism for identifying outliers using the specified
+ * {@link OutlierDetectingRule} and {@link OutlierDetector}.
+ */
+@UnstableApi
+public interface OutlierDetection {
```

Comment:
Optional) would `OutlierDetectorFactory` be a clearer name?

Reply:
I also thought about the name, `OutlierConfig` and `OutlierContext`.
`OutlierDetectorFactory` wasn't chosen since it had `OutlierDetectingRule`.

I'm not strong on the name. Let's listen to other folks' opinions and choose a better one.

Reply:
`OutlierDetection` sounds OK to me, but `OutlierDetectingRule` could be renamed to `OutlierRule`?

Reply:
> OutlierDetectorFactory wasn't chosen since it had OutlierDetectingRule.

It seems like `OutlierDetectingRule` is used alongside `OutlierDetector`, and both are retrieved and created from this `OutlierDetection`. What do you think of adding `OutlierDetectingRule` to `OutlierDetector` instead?
```
outlierDetector.rule().decide(...)
```